tinybytes/src/bytes_string.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Bytes;
+#[cfg(all(feature = "bytes_string", feature = "serde"))]
+use serde::ser::{Serialize, Serializer};
+use std::borrow::Borrow;
+use std::str::Utf8Error;
+
+#[cfg(feature = "bytes_string")]
+pub struct BufferWrapper<'a, 'b> {
+    buffer: Bytes,
+    pub underlying: &'a mut &'b [u8],
```

Comment:
Why do we need this - instead of a simple `fn as_slice(&self) -> &[u8]`? and possibly `unsafe fn as_mut_slice(&self) -> &mut [u8]`, forwarding to the Bytes instance?

Reply:
I'm not sure how that would work @bwoebi. The `underlying` is really `position` of the  message pack decoding, and the `buffer` should not be modified. Maybe the names should change to reflect this in some way? (Naming is hard™)